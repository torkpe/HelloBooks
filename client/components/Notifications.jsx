import React, { Component } from 'react';
import { connect } from 'react-redux';

import SingleNotification from './Notification.jsx';
import { getNotification } from '../actions/notification';
import { getAllBorrowed } from '../actions/history';

/**
 * @class Notifications
 * @classdesc returns Notificarions component
 */
class Notifications extends Component {
  constructor(props) {
    super(props);
    this.state = {
      borrows: [],
    };
  }
  /**
   * @returns {undefined}
   */
  componentWillMount() {
    const { id, star } = this.props.auth.user;
    this.props.getNotification(
      star,
      id
    );
  }
  /**
   * @param {object} nextProps
   * @return {undefined}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.borrowedBooks) {
      this.setState({
        borrows: nextProps.borrowedBooks,
      });
    }
  }
  /**
   * @return {XML} JSX
   */
  render() {
    const { notifications } = this.props;
    const mdlTableClass = `
    mdl-data-table
    mdl-js-data-table
    mdl-shadow--2dp
    `;
    return (
      <div className="mdl-grid">
        <div className="contents">
          <div className="ask">
            <h5>Notifications</h5>
          </div>
          {notifications && notifications.length > 0 ?
          <table className="table-bordered">
            {notifications.map(notification => (<SingleNotification
              key={notification.id}
              notification={notification}
              userId={this.props.auth.user.user}
              isAdmin={this.props.auth.user.isAdmin}
            />))}
          </table>
          :
          <div className="contents">
            <h5>You have no notification at this point in time</h5>
          </div>
          }
        </div>
      </div>
    );
  }
}
/**
 * @param {object} state
 * @return {object} props
 */
const mapStateToProps = state => ({
  notifications: state.getNotification.notifications,
  auth: state.auth,
  borrowedBooks: state.getAllBorrowed.borrowedBooks,
});
export default connect(mapStateToProps, {
  getNotification,
  getAllBorrowed,
})(Notifications);
