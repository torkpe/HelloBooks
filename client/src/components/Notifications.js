import React, { Component } from 'react';
import { connect } from 'react-redux';

import { getNotification } from '../actions/notification';

class Notifications extends Component {
    componentWillMount(){
        this.props.getNotification(this.props.auth.user.star)
    }
  render() {
      const notifications = this.props.notification.notifications
    return (
        <div className='mdl-grid'>                                               
            <div className='contents'>
                <div className="card-enlarge mdl-card mdl-shadow--3dp">                   
                    {notifications.map(notification => notification.message, <hr/>)
                    }
                </div>
            </div>
        </div>
    );
  }
}
const mapStateToProps = (state) => {
    return {
        notification: state.getNotification,
        auth: state.auth
    }
}
export default connect(mapStateToProps,{ getNotification })(Notifications);
