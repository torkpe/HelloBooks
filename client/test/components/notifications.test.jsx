import React from 'react';
import toJson from 'enzyme-to-json';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import render from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import { Notifications } from '../../components/Notifications.jsx';
import storeConfiguration from '../../store/store';

configure({ adapter: new Adapter() });

describe('<Notifications />', () => {
  const props = {
    getNotification: jest.fn(),
    auth: {
      user: {
        id: 1,
        star: 'bronze'
      }
    },
    notifications: [
      {
        message: 'fcjhkjhgcnbvnm,',
        bookId: 2
      },
      {
        message: 'fcjhkjhgcnbvnm,',
        bookId: 3
      },
      {
        message: 'fcjhkjhgcnbvnm,',
        bookId: 4
      },
    ],
  };
  let shallowComponent;
  beforeEach(() => {
    shallowComponent = shallow(<Notifications {...props} />);
  });
  it('renders <Notifications /> component', () => {
    const tree = toJson(shallowComponent);
    expect(tree).toMatchSnapshot();
  });
  it('checks if notifications is less than 0', () => {
    props.notifications = [];
    shallowComponent = shallow(<Notifications {...props} />);
    expect(shallowComponent.instance().props.notifications.length < 1).toEqual(true);
  });
});