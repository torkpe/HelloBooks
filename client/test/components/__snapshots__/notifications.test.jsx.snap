// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notifications /> renders <Notifications /> component 1`] = `
<div
  className="mdl-grid"
>
  <div
    className="contents"
  >
    <div
      className="ask"
    >
      <h5>
        Notifications
      </h5>
    </div>
    <table
      className="table-bordered"
    >
      <SingleNotification
        key="0"
        notification={
          Object {
            "bookId": 2,
            "message": "fcjhkjhgcnbvnm,",
          }
        }
      />
      <SingleNotification
        key="1"
        notification={
          Object {
            "bookId": 3,
            "message": "fcjhkjhgcnbvnm,",
          }
        }
      />
      <SingleNotification
        key="2"
        notification={
          Object {
            "bookId": 4,
            "message": "fcjhkjhgcnbvnm,",
          }
        }
      />
    </table>
  </div>
</div>
`;
